api/user/userController.js

```diff
@@ -1,6 +1,76 @@
 'use strict';
 
+var User = require('./userModel').User;
+var Friend = require('./userModel').Friend;
+
 exports.create = function (req, res) {
 
-   res.send("sad");
+    var friend = new Friend( { name: req.body.name });
+
+    var user = new User({
+        name: req.body.name,
+        friends: [friend]}
+    );
+
+    user.save(function (err,user) {
+        if (err) return handleError(err);
+         res.send(user);
+    });
+};
+
+exports.findUser = function (req, res) {
+
+    User.findById(req.params.id, function (err,user) {
+        if (err) return handleError(err);
+        res.json(user);
+    });
+};
+
+exports.addFriend = function (req, res) {
+
+    var newFriend = new Friend( { name: "test" });
```

Comment:
friend shouldn't be hardcoded.
read http://mongoosejs.com/docs/populate.html
and use real userIds for friends